### PR TITLE
Update the repository to use yarn instead of npm

### PR DESCRIPTION
**Pull request by emma-imber**
## What does this change?

Updated the repository to use yarn instead of npm, to be consistent with the rest of our code.

- Updated to yarn-lock instead of package-lock
- Removed the npmrc file as yarn enforces the engine for us without this
- Update the documentation to use yarn commands instead of npm commands
- Updated the scripts defined in package.json to use yarn
- Added a license to the package.json file (used the same one as commercial-core)

## How to test

Try running yarn or yarn install after cloning the repository and test that the commands work in yarn

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,14 +17,14 @@ found in the [`/legacy` folder](/legacy)
 
 ## Developing Locally
 
-Once you've created a project and installed dependencies with `npm install`,
+Once you've created a project and installed dependencies with `yarn install`,
 start a development server:
 
 ```bash
-npm run dev
+yarn run dev
```
Comment (Jakeii) [inline README.md:24]: you can omit the `run` with yarn :)

File: README.md [hunk 2]
```diff
@@ -17,14 +17,14 @@ found in the [`/legacy` folder](/legacy)
 
 ## Developing Locally
 
-Once you've created a project and installed dependencies with `npm install`,
+Once you've created a project and installed dependencies with `yarn install`,
```
Comment (Jakeii) [inline README.md:20]: `install` isn't needed with yarn

**Review by Jakeii — APPROVED** [review]
yarn ftw!

File: README.md (hunk 1 above)
Comment (emma-imber) [reply README.md]: Oh fab, thanks!

File: package.json [hunk 3]
```diff
@@ -1,11 +1,12 @@
 {
 	"name": "@guardian/commercial-templates-builder",
 	"version": "0.0.1",
+	"license": "Apache-2.0",
 	"scripts": {
 		"dev": "NODE_ENV=development vite dev --port 7777",
 		"build": "vite build && touch build/.nojekyll",
 		"publish": "npx gh-pages --dist build --dotfiles",
```
Comment (arelra) [inline package.json:8]: Think we can also replace `npx` with `yarn` but please verify!

**Review by arelra — APPROVED** [review]
Thanks for tidying this up. Yay for standardisation! 🎉

File: package.json (hunk 3 above)
Comment (emma-imber) [reply package.json]: Just checked and you're right, thanks Ravi!

**Pull request merged**